numpyro/infer/mcmc.py

```diff
@@ -491,11 +492,25 @@ def get_extra_fields(self, group_by_chain=False):
         return {k: v for k, v in states.items() if k != self._sample_field}
 
     def print_summary(self, prob=0.9, exclude_deterministic=True):
+        """
+        Print the statistics of posterior samples collected during running this MCMC instance.
+
+        :param float prob: the probability mass of samples within the credible interval.
+        :param bool exclude_deterministic: whether or not print out the statistics
+            at deterministic sites.
+        """
         # Exclude deterministic sites by default
         sites = self._states[self._sample_field]
         if isinstance(sites, dict) and exclude_deterministic:
-            sites = {k: v for k, v in self._states[self._sample_field].items()
-                     if k in self._last_state.z}
+            state_sample_field = attrgetter(self._sample_field)(self._last_state)
+            # XXX: there might be the case that state.z is not a dictionary but
+            # its postprocessed value `sites` is a dictionary.
+            # TODO: in general, when both `sites` and `state.z` are dictionaries,
+            # they can have different key names, not necessary due to deterministic
```

Comment:
Could you explain this further - when will `sites` and `state.z` have different names?

Reply:
This depends on `post_processing` code, e.g. `state.z = {'ABC_b': 1}` while `sites = {'b': 1}`. The below code will return an empty dict because it thinks `'b'` is an deterministic site.

Reply:
LGTM, but I am thinking that we should keep track of deterministic and latent nodes separately so that we don't need to bother with all this downstream. We can refactor this later.